### PR TITLE
Fix popularity channel errors

### LINKED ISSUE #3715: Error in the popularity community
I got the following stack trace:

```
Traceback (most recent call last):
  File "/Users/martijndevos/Documents/tribler/Tribler/pyipv8/ipv8/deprecated/community.py", line 330, in on_packet
    self.decode_map[data[22]](source_address, data)
  File "/Users/martijndevos/Documents/tribler/Tribler/community/popularity/community.py", line 104, in on_torrent_info_request
    self.send_torrent_info_response(payload.infohash, peer=peer)
  File "/Users/martijndevos/Documents/tribler/Tribler/community/popularity/community.py", line 211, in send_torrent_info_response
    packet = self.create_message_packet(MSG_TORRENT_INFO_RESPONSE, info_response)
  File "/Users/martijndevos/Documents/tribler/Tribler/community/popularity/pubsub.py", line 188, in create_message_packet
    payload = payload if isinstance(payload, list) else payload.to_pack_list()
  File "/Users/martijndevos/Documents/tribler/Tribler/community/popularity/payload.py", line 150, in to_pack_list
    ('varlenH', str(self._name)),
UnicodeEncodeError: 'ascii' codec can't encode character u'\u7231' in position 16: ordinal not in range(128)
```

### LINKED ISSUE #3716: Error in popularity community when sending messages
I got the following stack trace:

```
Jul 05 15:34:39 tribler-runner1 twistd[27982]: Traceback (most recent call last):
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/pyipv8/ipv8/deprecated/community.py", line 330, in on_packet
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     self.decode_map[data[22]](source_address, data)
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/community/popularity/community.py", line 104, in on_torrent_info_request
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     self.send_torrent_info_response(payload.infohash, peer=peer)
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/community/popularity/community.py", line 211, in send_torrent_info_response
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     packet = self.create_message_packet(MSG_TORRENT_INFO_RESPONSE, info_response)
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/community/popularity/pubsub.py", line 189, in create_message_packet
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     return self._ez_pack(self._prefix, message_type, [auth, dist, payload])
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/pyipv8/ipv8/deprecated/community.py", line 100, in _ez_pack
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     packet += self.serializer.pack_multiple(format_list)
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/pyipv8/ipv8/messaging/serialization.py", line 191, in pack_multiple
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     out += self.pack(*packable)
Jul 05 15:34:39 tribler-runner1 twistd[27982]:   File "/opt/tribler/Tribler/pyipv8/ipv8/messaging/serialization.py", line 177, in pack
Jul 05 15:34:39 tribler-runner1 twistd[27982]:     return self._packers[format].pack(*data)
Jul 05 15:34:39 tribler-runner1 twistd[27982]: PackError: Could not pack item 2: ('I', None)
Jul 05 15:34:39 tribler-runner1 twistd[27982]: error: cannot convert argument to integer
```

### PR DESCRIPTION
**Pull request by xoriole**
Fixes popularity channel issues.
Fix #3716 pack issue
Fix #3715 unicode issue

### CODE REVIEW

File: Tribler/Core/Utilities/random_utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import string
+
```
Comment (qstokkink) [inline Tribler/Core/Utilities/random_utils.py]: Please remove this space.

File: Tribler/Core/Utilities/random_utils.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return ''.join(random.choice(chars) for _ in range(size))
+
+
+def random_infohash():
+    """ Generates a random torrent infohash string """
+    return ''.join(random.choice('0123456789abcdef') for _ in range(20))
+
+
+def random_unicode(length=6):
+    """ Generates a random unicode string """
```
Comment (qstokkink) [inline Tribler/Core/Utilities/random_utils.py]: This actually not random unicode, but random valid utf-8.

File: Tribler/Core/Utilities/random_utils.py (hunk 1 above)
Comment (devos50) [reply Tribler/Core/Utilities/random_utils.py]: Also, the order of the `string` and `random` imports should be reverted.

**Comment by qstokkink** [comment]
@xoriole is this ready?

**Comment by xoriole** [comment]
Yes, but I still want to use this to check the Jenkins & sonar.

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Pull request merged**